### PR TITLE
Support for Kubernetes v1.29

### PR DESCRIPTION
**Pull request by acumino**
**How to categorize this PR?**

/area open-source
/kind enhancement

**What this PR does / why we need it**:
Support for Kubernetes v1.29.

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/8871

**Special notes for your reviewer**:
- Code changes have been tested in the following scenarios:
  - Create/Delete a new (normal/workerless) Shoot cluster with 1.29.0
  - Upgrade an existing (normal/workerless) Shoot cluster to 1.29.0

**Release note**:

```feature operator
Gardener can now support clusters with Kubernetes version 1.29. To allow creation/update of 1.29 clusters you will have to update the version of your provider extension(s) to a version that supports 1.29 as well. Please consult the respective releases and notes in the provider extension's repository.
```

```feature developer
Gardener can now support clusters with Kubernetes version 1.29. Extension developers have to prepare individual extensions as well to work with 1.29.
```

### CODE REVIEW

**Review by ary1992 — COMMENTED** [review]
Thanks for the PR.
Just one comment otherwise lgtm.
Please add `flowcontrol.apiserver.k8s.io/v1` under https://github.com/gardener/gardener/blob/71e5bbc96f61d5ff83ce3be328238392ed3a564d/pkg/utils/validation/apigroups/apigroups.go#L30

**Comment by shafeeqes** [comment]
Thanks!
/approve

**Review by ary1992 — COMMENTED** [review]
/lgtm

File: pkg/component/shootsystem/shootsystem.go [hunk 1]
```diff
@@ -355,10 +355,21 @@ func (s *shootSystem) getServiceAccountNamesToInvalidate() []string {
...
 			"validatingadmissionpolicy-status-controller",
 		)
 	}
 
+	if versionutils.ConstraintK8sGreaterEqual129.Check(s.values.KubernetesVersion) {
+		kubeControllerManagerServiceAccountNames = append(kubeControllerManagerServiceAccountNames,
+			"service-cidrs-controller",
+		)
+	}
+
```
Comment (shafeeqes) [inline pkg/component/shootsystem/shootsystem.go:372]: Please add a unit test for this.

**Review by shafeeqes — CHANGES_REQUESTED** [review]
One minor thing

**Pull request merged**